test/integration/esys-macsequencestart.int.c

```diff
@@ -0,0 +1,289 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*******************************************************************************
+ * Copyright 2017-2018, Fraunhofer SIT sponsored by Infineon Technologies AG
+ * All rights reserved.
+ *******************************************************************************/
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdlib.h>
+
+#include "tss2_esys.h"
+
+#include "esys_iutil.h"
+#define LOGMODULE test
+#include "util/log.h"
+#include "util/aux_util.h"
+
+/** Test the ESYS commands: MAC_Start, SequenceUpdate, and SequenceComplete.
+ *
+ * The MAC key is created by using Esys_CreatePrimary.
+ *
+ * Tested ESYS commands:
+ *  - Esys_CreatePrimary() (M)
+ *  - Esys_FlushContext() (M)
+ *  - Esys_MAC_Start() (M)
+ *  - Esys_SequenceComplete() (M)
+ *  - Esys_SequenceUpdate() (M)
+ *  - Esys_StartAuthSession() (M)
+ *
+ * Used compiler defines: TEST_SESSION
+ *
+ * @param[in,out] esys_context The ESYS_CONTEXT.
+ * @retval EXIT_FAILURE
+ * @retval EXIT_SUCCESS
+ */
+
+int
+test_esys_macsequencestart(ESYS_CONTEXT * esys_context)
+{
+    TSS2_RC r;
+    ESYS_TR primaryHandle = ESYS_TR_NONE;
+
+    TPM2B_PUBLIC *outPublic = NULL;
+    TPM2B_CREATION_DATA *creationData = NULL;
+    TPM2B_DIGEST *creationHash = NULL;
+    TPMT_TK_CREATION *creationTicket = NULL;
+
+    TPM2B_DIGEST *result = NULL;
+    TPMT_TK_HASHCHECK *validation = NULL;
+
+#ifdef TEST_SESSION
+    ESYS_TR session = ESYS_TR_NONE;
+    TPMT_SYM_DEF symmetric = {.algorithm = TPM2_ALG_AES,
+                              .keyBits = {.aes = 128},
+                              .mode = {.aes = TPM2_ALG_CFB}
+    };
+    TPMA_SESSION sessionAttributes;
+    TPM2B_NONCE nonceCaller = {
+        .size = 20,
+        .buffer = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+                   11, 12, 13, 14, 15, 16, 17, 18, 19, 20}
+    };
+
+    memset(&sessionAttributes, 0, sizeof sessionAttributes);
```

Comment:
Can't we just `TPMA_SESSION sessionAttributes = 0;` since TPMA_SESSION is a UINT8?

Reply:
@malikabhi05 resolve this.